src/main/java/jenkins/branch/MultiBranchProject.java

```diff
@@ -2011,19 +2006,20 @@ private void observeProject(@NonNull SCMHead head, @NonNull SCMRevision revision
                 // the previous "revision" for this head is not a revision for the current source
                 // either because the head was removed and then recreated, or because the head
                 // was taken over by a different source, thus the previous revision is null
-                doAutomaticBuilds(head, revision, rawName, project, revisionActions, null, null);
+                doAutomaticBuilds(head, revision, rawName, project,null, null);
             } else {
                 // get the previous revision
                 SCMRevision scmLastBuiltRevision = _factory.getRevision(project);
 
                 if (isChangesDetected(revision, project, scmLastBuiltRevision)) {
+                    listener.getLogger()
+                            .format("Changes detected: %s (%s → %s)%n", rawName, scmLastBuiltRevision, revision);
```

Comment:
Moved this up to make it extra clear that is doesn't use `scmLastSeenRevision `.